### DISCUSSION TITLE
How to not replace the "default message toolbar" of thunderbird with the "conversations UI"?

### DISCUSSION THREAD

**Discussion by neoragex80** [protz/thunderbird-conversations#1963] [General]
I have a lot of useful addons activated which are accessible through buttons on the default toolbar of the message pane. Your Conversations addon is great, but overwrites that whole toolbar completely and none of my addons are usable anymore. This kept me for years from using the Converations addon. How to change that? Thank you....

**Comment by Standard8** [comment]
The problem with the "default toolbar" of the message pane is that it is oriented to a single message. With Conversations, there's different levels of toolbars.

This would probably require Conversations somehow hooking into the extension code to learn about other add-ons and insert buttons into the appropriate places.

#1665 is the issue that covers this.

**Reply by neoragex80** [reply]
Hi,
thank you for your reply...

Is it not possible to either duplicate the default tb message toolbar for every single conversations message pane or to get the configured addon buttons of the default tb message toolbar and display them on every conversations toolbar?

It's just that you force the users to make a decision - to either use conversations and loose all other addon functionality or to abandon conversations although it's cool.

Seems to me that you should really overthink that...

Best,
Chris

**Reply by Standard8** [reply]
> Is it not possible to either duplicate the default tb message toolbar for every single conversations message pan

Unfortunately not, it doesn't quite work like that.

> or to get the configured addon buttons of the default tb message toolbar and display them on every conversations toolbar?

That's effectively what I was suggesting previously.

> It's just that you force the users to make a decision - to either use conversations and loose all other addon functionality or to abandon conversations although it's cool.
> 
> Seems to me that you should really overthink that...

Unfortunately I don't have time to get to everything. I would love to fix some of the add-on compat issues, but it is a hard issue that's probably going to take several hours to fix. There's also plenty of maintenance to do to keep Conversations going and fix all the other bugs that happen.
